include/secp256k1.h

```diff
@@ -169,6 +169,20 @@ typedef int (*secp256k1_nonce_function)(
 #  define SECP256K1_ARG_NONNULL(_x)
 # endif
 
+/** Attribute for marking functions, types, and variables as deprecated
+  * GCC <4.5 supports the attribute but not with an optional message argument. */
+#if !defined(SECP256K1_BUILD)
+# if defined(__GNUC__) && SECP256K1_GNUC_PREREQ(4, 5)
+#  define SECP256K1_DEPRECATED(_msg) __attribute__ ((__deprecated__(_msg)))
+# elif defined(__has_attribute)
+#  if __has_attribute(__deprecated__)
+#   define SECP256K1_DEPRECATED(_msg) __attribute__ ((__deprecated__(_msg)))
```

Comment:
This line seems to be identical to the one in the previous branch. Did you intend to remove the `_msg` argument here?

Reply:
No this is intentional, I just didn't find a nicer way to write it...

Let me explain. One can't write this as a single condition
```
(defined(__GNUC__) && SECP256K1_GNUC_PREREQ(4, 5)) || (__has_attribute(__deprecated__) && __has_attribute(__deprecated__)
```
because this is illegal if `__has_attribute` is really not defined.

The `__has_attribute` is actually a clang thing (plus very recent GCC) and the clang docs suggest this here:
```
#ifndef __has_attribute
  #define __has_attribute(x) 0
#endif
```

Looks nice but I'm somewhat hesitant to define a reserved `__` identifier in a public include file. It's technically UB and in this case it could really lead to strange issues: Say the user includes our file and then checks `defined(__has_attribute)` in their code... I don't see an immediate problem with this, but it seems just wrong to do this.

Reply:
Oh indeed, "very recent GCC" is not true. I switched to your suggestion (also added a missing #else case)